pkgs/development/compilers/chez-racket/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv, lib, fetchFromGitHub
+, coreutils, cctools
+, ncurses, libiconv, libX11
+}:
+
+stdenv.mkDerivation rec {
+  pname = "chez-scheme-racket";
+  version = "unstable-2021-07-09";
+
+  src = fetchFromGitHub {
+    owner  = "racket";
+    repo   = "ChezScheme";
+    rev    = "f1f2294be7809bbc15c05af6f7a99e2b47070c18";
+    sha256 = "wXjPyTKqIifRKoeBOn+G+pq9QK/1suTe5VqqZNanSOI=";
+    fetchSubmodules = true;
```

Comment:
We don't want to use the vendored lz4 or zlib and preferable none of the other libs, too.

Reply:
I took a look at it and it does not seem trivial unfortunately.

Reply:
You can pass `configure` your shared zlib and lz4: https://github.com/racket/ChezScheme/blob/91a33d86e8e5f637bbc96afa765aff6b3b66120c/configure#L320-L332

Nanopass and STeX are not really shared libraries and are only used in the build process, but in Guix we share their origin definitions with the upstream/Cisco Chez Scheme package.

Reply:
Thanks for the help!